app/scripts/lib/offscreen-bridge/ledger-offscreen-bridge.ts

```diff
@@ -0,0 +1,173 @@
+import { LedgerBridge } from '@metamask/eth-ledger-bridge-keyring';
+import {
+  LedgerAction,
+  OffscreenCommunicationEvents,
+  OffscreenCommunicationTarget,
+} from '../../../../shared/constants/offscreen-communication';
+
+/**
+ * This class is used as a custom bridge for the Ledger connection. Every
+ * hardware wallet keyring also requires a bridge that has a known interface
+ * that the keyring can call into for specific functions. The bridge then makes
+ * whatever calls or requests it needs to in order to fulfill the request from
+ * the keyring. In this case, the bridge is used to communicate with the
+ * Offscreen Document. Inside the Offscreen document the ledger-iframe is
+ * embedded that will listen for these calls and communicate with the
+ * ledger device via the ledger keyring iframe.
+ */
+export class LedgerOffscreenBridge implements LedgerBridge {
+  isDeviceConnected = false;
+
+  init() {
+    chrome.runtime.onMessage.addListener((msg) => {
+      if (
+        msg.target === OffscreenCommunicationTarget.extension &&
+        msg.event === OffscreenCommunicationEvents.ledgerDeviceConnect
+      ) {
+        this.isDeviceConnected = true;
+      }
+    });
+
+    return Promise.resolve();
+  }
+
+  destroy() {
+    // TODO: remove listener
+    return Promise.resolve();
+  }
+
+  attemptMakeApp() {
+    return new Promise<boolean>((resolve, reject) => {
+      chrome.runtime.sendMessage(
+        {
+          target: OffscreenCommunicationTarget.ledgerOffscreen,
+          action: LedgerAction.makeApp,
+        },
+        (response) => {
+          if (response.success) {
```

Comment:
I notice this if else block is different from what is currently on the main branch on the eth-ledger-bridge-keyring repo. Is that expected?

https://github.com/MetaMask/eth-ledger-bridge-keyring/blob/main/src/ledger-iframe-bridge.ts#L132-L138

Reply:
Its mostly expected, I will add an else if error with a catch for if neither success or error is defined.

Reply:
replicated the else catch for a missing error